app/src/org/commcare/views/EntityDetailView.java

```diff
@@ -255,10 +263,12 @@ public void onClick(View v) {
         } else if (FORM_GRAPH.equals(form) && field instanceof GraphData) {    // if graph parsing had errors, they'll be stored as a string

```

Comment:
At some point in the future can you breakup this 100-line graph specific code block into a series of helper methods?